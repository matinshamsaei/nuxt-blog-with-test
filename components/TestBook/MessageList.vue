<template>
  <div>
    <ul>
      <message v-for="message in messages" :key="message">
        {{ message }}
      </message>
    </ul>
    <div class="matin">
      <span>content</span>
    </div>
  </div>
</template>

<script>
import Message from "@/components/TestBook/Message.vue";

export default {
  name: "list",
  props: ["messages"],
  components: {
    Message
  }
};
</script>
