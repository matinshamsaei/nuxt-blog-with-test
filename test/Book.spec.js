import { mount } from "@vue/test-utils";
import MessageList from "@/components/TestBook/MessageList";
import Message from "@/components/TestBook/Message";

// WITHOUT VUE-TEST-UTILS
// describe("App.test.js", () => {
//     let cmp, vm;
//     beforeEach(() => {
//         cmp = Vue.extend(App); // Create a copy of the original component
//         vm = new cmp({
//             data: {
//                 // Replace data value with this fake data
//                 messages: ["Cat"]
//             }
//         }).$mount(); // Instances and mounts the component
//     });
//     it('equals messages to ["Cat"]', () => {
//         expect(vm.messages).toEqual(["Cat"]);
//     });
// });

// WITH VUE-TEST-UTILS JUST MESSAGELIST CMP
describe("App.test.js", () => {
    let cmp;

    beforeEach(() => {
        cmp = mount(MessageList, {
            // Create a copy of the original component
            propsData: {
                // Replace data value with this fake data
                messages: ["Cat"]
            }
        });
    });

    it('has received ["Cat"] as the message property', () => {
        expect(cmp.vm.messages).toEqual(["Cat"]);
    });

    it("has the expected html structure", () => {
        expect(cmp.element).toMatchSnapshot();
    });

    it("div span text", () => {
        // cmp.find(".matin span").element  ===>  <span>content</span>
        expect(cmp.find(".matin span").text()).toEqual("content")
    })

    it("Message cmp is exist", () => {
        expect(cmp.findComponent(Message).exists()).toBe(true)
        // expect(cmp.find(".message").exists()).toBe(true)
    })

    it("Message cmp has message class", () => {
        // expect(cmp.findComponent(Message).attributes().class).toBe("message")
        expect(cmp.findComponent(Message).classes()).toContain("message")
    })

    it("Message cmp has style margin-top: 10px", () => {
        expect(cmp.findComponent(Message).attributes().style).toBe("margin-top: 10px;")
    })
});

