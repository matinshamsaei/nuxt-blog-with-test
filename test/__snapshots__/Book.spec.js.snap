// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.test.js has the expected html structure 1`] = `
<div>
  <ul>
    <li
      class="message"
      style="margin-top: 10px;"
    >
      
    </li>
  </ul>
   
  <div
    class="matin"
  >
    <span>
      content
    </span>
  </div>
</div>
`;
